src/Evolution/Systems/NewtonianEuler/Characteristics.hpp

```diff
@@ -0,0 +1,159 @@
+ * \qquad
+ * (t_2)^i = \left( \begin{array}{c}
+ * -\sin\theta_y \cos\theta_z\\ -\sin\theta_y \sin\theta_z\\ \cos\theta_y
+ * \end{array}\right).
+ * \f}
+ *
+ * The angles are obtained from the components of the normal vector.
+ * For the rotation around the \f$y-\f$axis,
+ * \f{align*}
+ * \sin\theta_y &= n_z,\\
+ * \cos\theta_y &= \sqrt{1 - n_z^2} = \sqrt{n_x^2 + n_y^2},
+ * \f}
+ * where the second expression is preferred for the cosine, as
+ * it is better behaved for \f$|n_z|\approx 1\f$. Also, since
+ * \f$|n_z| \leq 1\f$, \f$|\theta_y| \leq \pi/2\f$, so only the positive
+ * root is needed. For the rotation around the \f$z-\f$axis,
+ * \f{align*}
+ * \sin\theta_z = \frac{n_y}{\sqrt{n_x^2 + n_y^2}},\\
+ * \cos\theta_z = \frac{n_x}{\sqrt{n_x^2 + n_y^2}}.
+ * \f}
+ * These formulas are valid except when \f$n_x = n_y = 0\f$. In the latter
+ * case, \f$\sin\theta_z = 0\f$ and \f$\cos\theta_z = 1\f$ are imposed.
+ */
+template <size_t Dim>
+void right_eigenvectors(
+    gsl::not_null<tnsr::ij<DataVector, Dim + 2, Frame::State>*>
+        right_eigenvectors,
+    const Scalar<DataVector>& mass_density,
+    const Scalar<DataVector>& energy_density,
+    const tnsr::I<DataVector, Dim>& velocity,
+    const Scalar<DataVector>& pressure, const Scalar<DataVector>& zeta,
+    const Scalar<DataVector>& sound_speed_squared,
+    const tnsr::i<DataVector, Dim>& normal) noexcept;
+
+/*!
+ * \brief Compute the left eigenvectors of NewtonianEuler system
+ *
+ *  The eigenvectors are the rows of the \f$(\text{Dim} + 2)\times
+ * (\text{Dim} + 2)\f$ matrix
+ *
+ * \f{align*}
+ * \Omega_L = \left( \begin{array}{ccc}
+ * \dfrac{\zeta\Theta + c_s v_n}{2c_s^2} & -\dfrac{\zeta v_i + c_sn_i}{2c_s^2} &
+ * \dfrac{\zeta}{2c_s^2}\\
+ * -v_1 & (t_1)_i & 0\\
+ * -v_2 & (t_2)_i & 0\\
+ * \dfrac{h - v^2}{c_s^2} & \dfrac{v_i}{c_s^2} & -\dfrac{1}{c_s^2}\\
+ * \dfrac{\zeta\Theta - c_s v_n}{2c_s^2} & -\dfrac{\zeta v_i - c_sn_i}{2c_s^2} &
+ * \dfrac{\zeta}{2c_s^2}
+ * \end{array}\right),
+ * \f}
+ *
+ * where \f$i = 1,...,\text{Dim}\f$, \f$v^i\f$ is the velocity, \f$n^i\f$
+ * is a unit normal vector, \f$v_n = n_i v^i\f$ is the velocity projected
+ * onto the normal, \f$\zeta = \kappa/\rho\f$ with
+ * \f$\kappa = (\partial p/\partial\epsilon)_\rho\f$, \f$\zeta \Theta = \chi +
+ * \zeta(v^2 - e/\rho)\f$ with \f$\chi = (\partial p/\partial\rho)_\epsilon,
+ * \f$ \f$\rho\f$ is the mass density, \f$e\f$ is the energy density,
+ * \f$p\f$ is the pressure, \f$\epsilon\f$ is the specific
+ * internal energy, \f$h\f$ is the specific enthalpy, \f$c_s\f$ is the
+ * sound speed, and \f$(t_1)^i\f$ and \f$(t_2)^i\f$ are two unit vectors
+ * tangent to the plane orthogonal to \f$n^i\f$. The tangent vectors are
+ * computed from the normal vector in the same way as in
+ * right_eigenvectors().
+ *
+ * Note that \f$\Omega_L = \Omega_R^{-1}\f$, \f$\Omega_R\f$ being the
+ * matrix whose columns are the right_eigenvectors().
+ */
+template <size_t Dim>
+void left_eigenvectors(
```

Comment:
We dont think you need to have the left and right eigenvectors coded up explicitly, things like the marquina flux should be specialized to the system